File: sistema-educativo-frontend/src/services/tramiteService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/tramites';

const tramiteService = {
    obtenerTramites: ()  => axios.get(API_URL),
    obtenerTramitePorId: (id) => axios.get(`${API_URL}/${id}`),
    creatTramite: (tramite) => axios.post(API_URL, tramite),
    actualizarTramite: (id, tramite) => axios.put(`${API_URL}/${id}`, tramite),
    eliminarTramite: (id) => axios.delete(`${API_URL}/${id}`)
};

export default tramiteService;

File: sistema-educativo/src/main/java/com/educacion/service/TramiteService.java
package com.educacion.service;

import com.educacion.model.Tramite;
import com.educacion.repository.TramiteRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class TramiteService {

    @Autowired
    private TramiteRepository tramiteRepository;

    public List<Tramite> obtenerTramites() {
        return tramiteRepository.findAll();
    }

    public ResponseEntity<Tramite> obtenerTramitePorId(String id) {
        return tramiteRepository.findById(id).map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

    public Tramite crearTramite(Tramite tramite) {
        return tramiteRepository.save(tramite);
    }

    public ResponseEntity<Tramite> actualizarTramite(String id, Tramite tramite) {
        return tramiteRepository.findById(id).map(t -> {
            t.setDescripcion(tramite.getDescripcion());
            t.setEstatus(tramite.getEstatus());
            return ResponseEntity.ok(tramiteRepository.save(t));
        }).orElse(ResponseEntity.notFound().build());
    }

    public ResponseEntity<Object> eliminarTramite(String id) {
        return tramiteRepository.findById(id).map(t -> {
            tramiteRepository.delete(t);
            return ResponseEntity.ok().build();
        }).orElse(ResponseEntity.notFound().build());
    }
}
